### ISSUE TITLE
Use `GIT_STRATEGY` to speed up GitLab pipelines

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-django-template#415]
There are different strategies to be used `fetch`, `clone`, `none`.
Some of them are faster than default `clone`. So, we need to find out if that's actually is going to speed up the whole pipeline.

**Comment by ProSerg** [comment]
I think that the Docker executor of gitlab runner by default uses `fetch`. From this point of view it is difficult to accelerate process here.

**Comment by sobolevn** [comment]
Here's the logs you were asking about.

## First pipeline: test

```
Running with gitlab-runner 11.0.0 (5396d320)
  on docker-auto-scale fa6cab46
Using Docker executor with image docker:latest ...
Starting service docker:dind ...
Pulling docker image docker:dind ...
Using docker image sha256:a340e9c87a9562c063519ce3c6a80e315932107e4a56a5cb17adb5dd5658ecd4 for docker:dind ...
Waiting for services to be up and running...
Pulling docker image docker:latest ...
Using docker image sha256:5d152d55326b2c48923246c28605dd7ffb4813af9624b18f97db9934a47026cb for docker:latest ...
Running on runner-fa6cab46-project-4875972-concurrent-0 via runner-fa6cab46-srm-1531344406-5c585ad1...
Cloning repository...
Cloning into /builds/wemake.services/...
Checking out 37fe7c5b as master...
Skipping Git submodules setup
$ apk update && apk upgrade
```

## Second pipeline: build

```
Running with gitlab-runner 11.0.0 (5396d320)
  on docker-auto-scale 0277ea0f
Using Docker executor with image docker:latest ...
Starting service docker:dind ...
Pulling docker image docker:dind ...
Using docker image sha256:a340e9c87a9562c063519ce3c6a80e315932107e4a56a5cb17adb5dd5658ecd4 for docker:dind ...
Waiting for services to be up and running...
Pulling docker image docker:latest ...
Using docker image sha256:5d152d55326b2c48923246c28605dd7ffb4813af9624b18f97db9934a47026cb for docker:latest ...
Running on runner-0277ea0f-project-4875972-concurrent-0 via runner-0277ea0f-srm-1531344957-c4589dd9...
Cloning repository...
Cloning into /builds/wemake.services/...
Checking out 37fe7c5b as master...
Skipping Git submodules setup
$ apk update && apk upgrade
```

@ProSerg so, does it make any sence to you?

**Comment by ProSerg** [comment]
It doesn't change behavior of process. I tried to put GIT_STARTEGY fetch, but the repository all the same cloned the source code.

Also I tried to cache pip files, but it to a privel to increase in time of performance of assembly for 2 minutes.

May me create a preliminary image where would be saveing result of performance:

`pipenv install $(test "$DJANGO_ENV" == production || echo "--dev") --deploy --system --ignore-pipfile`
